### PR TITLE
[add] utest kernel mutex_tc.c file

### PR DESCRIPTION
**Pull request by tyustli**
## 拉取/合并请求描述：(PR description)

[
- add mutex testcase config and source file

- test platform:qemu-a9

- result:

image

]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

File: examples/utest/testcases/kernel/mutex_tc.c [hunk 1]
```diff
@@ -0,0 +1,676 @@
...
+    }
+
+    result = rt_mutex_delete(dynamic_mutex);
+    if (RT_EOK != result)
+    {
+        uassert_true(RT_FALSE);
+    }
+
+    /* FIFO mode */
+    dynamic_mutex = rt_mutex_create("dynamic_mutex", RT_IPC_FLAG_FIFO);
```
Comment (mysterywolf) [inline examples/utest/testcases/kernel/mutex_tc.c:351]: mutex 的 FIFO 还需要测吗？ @Guozhanxin

Comment (Guozhanxin) [reply examples/utest/testcases/kernel/mutex_tc.c:351]: 可以测试FIFO的行为和 PRIO 的行为是否一致。内部已经忽略了 FIFO的模式，都是 PRIO的模式

Comment (mysterywolf) [reply examples/utest/testcases/kernel/mutex_tc.c:351]: 嗷嗷

File: examples/utest/testcases/kernel/mutex_tc.c [hunk 2]
```diff
@@ -0,0 +1,676 @@
...
+                                       2048,
+                                       10,
+                                       10);
+    if (RT_NULL == tid)
+    {
+        uassert_true(RT_FALSE);
+        return;
+    }
+
+    /* startup thread adn take mutex second */
```
Comment (Guozhanxin) [inline examples/utest/testcases/kernel/mutex_tc.c:470]: 你这里是不是想写 and

File: examples/utest/testcases/kernel/mutex_tc.c [hunk 3]
```diff
@@ -0,0 +1,676 @@
...
+    tid3 = RT_NULL;
+
+    dynamic_mutex = rt_mutex_create("dynamic_mutex", RT_IPC_FLAG_PRIO);
+    if (RT_NULL == dynamic_mutex)
+    {
+        uassert_true(RT_FALSE);
+        return;
+    }
+
+    /* 创建线程 1 */
```
Comment (Guozhanxin) [inline examples/utest/testcases/kernel/mutex_tc.c:602]: 中文注释删掉吧

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**